associated-token-account/program/src/lib.rs

```diff
@@ -40,7 +40,7 @@ pub fn get_associated_token_address(
     get_associated_token_address_with_program_id(
         wallet_address,
         token_mint_address,
-        &spl_token::id(),
+        &spl_token_2022::id(),
```

Comment:
🤔 do we really want to do this?  It'll essentially forcibly upgrade all downstream consumers to token-2022 if they upgrade their ATA crate version.  Maybe yes, but if so I think that's a major ATA version bump.  My gut is no for now, to require downstream to deliberately opt-in to token-2022 in all cases until it becomes ubiquitous.

Reply:
oops, nope! This is a holdover from before, made clear by the spl-token -> spl-token-2022 crate rename.